src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs

```diff
@@ -8,29 +8,43 @@ namespace System.Data.SqlClient
 {
     internal sealed class TdsParserStateObjectFactory
     {
-        public static bool useManagedSni = false;
+
+        private const string UseLegacyNetworkingOnWindows = "System.Data.SqlClient.UseLegacyNetworkingOnWindows";
 
         public static readonly TdsParserStateObjectFactory Singleton = new TdsParserStateObjectFactory();
 
+        static TdsParserStateObjectFactory()
+        {
+            bool shouldUseLegacyNetorking;
+            // If The appcontext switch is set then Use Managed SNI based on the value. Otherwise Managed SNI should always be used.
+            UseManagedSni = AppContext.TryGetSwitch(UseLegacyNetworkingOnWindows, out shouldUseLegacyNetorking) ? !shouldUseLegacyNetorking : true;
+        }
+
+        public static bool UseManagedSni
```

Comment:
Looking at the code that has been added for SNI in Managed SNI, I think it should be capitalized. I will change this.